aot-std-optimizers/src/main/java/io/micronaut/aot/std/sourcegen/AbstractStaticServiceLoaderSourceGenerator.java

```diff
@@ -91,19 +93,34 @@ public void generate(@NonNull AOTContext context) {
             serviceNames = context.getConfiguration().stringList(findOption(this.getClass(), SERVICE_TYPES).key());
         }
         if (substitutions == null) {
-            var resourceNames = new LinkedHashSet<String>();
-            resourceNames.add("application");
-            context.getAnalyzer().getEnvironmentNames()
+            var environmentNames = new LinkedHashSet<>(context.getAnalyzer().getEnvironmentNames());
+            environmentNames.addAll(
+                context.getConfiguration().stringList(Environments.POSSIBLE_ENVIRONMENTS_NAMES)
                 .stream()
-                .map(env -> "application-" + env)
-                .forEach(resourceNames::add);
-            context.getConfiguration().stringList(Environments.POSSIBLE_ENVIRONMENTS_NAMES)
-                .stream()
-                .filter(env -> !"default".equals(env))
-                .map(env -> "application-" + env)
-                .forEach(resourceNames::add);
+                .filter(env -> !"default".equals(env)).toList()
+            );
             substitutions = new HashMap<>();
+            if (context.getConfiguration().isFeatureEnabled(GenericPropertySourceGenerator.ID)) {
+                List<ActiveEnvironment> environments = new ArrayList<>();
+                int i = 0;
+                for (String name: environmentNames) {
+                    environments.add(ActiveEnvironment.of(name, i));
+                    ++i;
+                }
+
+                var propGen = new GenericPropertySourceGenerator(context, environments);
+                propGen.generate(context);
+                if (MetadataUtils.isEnabledOn(context.getRuntime(), propGen)) {
+                    LOGGER.debug("Substituting {} with {}", PropertySourceLoader.class.getName(), propGen.getClass().getName());
+                    substitutions.put(YamlPropertySourceLoader.class.getName(), propGen);
+                }
+            }
             if (context.getConfiguration().isFeatureEnabled(YamlPropertySourceGenerator.ID)) {
+                LOGGER.warn("The " + YamlPropertySourceGenerator.ID + " is deprecated in favor of " + GenericPropertySourceGenerator.ID);
+                var resourceNames = new LinkedHashSet<String>();
+                resourceNames.add(Environment.DEFAULT_NAME);
+                environmentNames.stream().map(e -> Environment.DEFAULT_NAME + "-" + e).forEach(resourceNames::add);
+
```

Comment:
I did not actually remove the old generator. If the feature is enabled, the generator will run. I did add a warning that it is deprecated.

Could be a good idea to remove it and add incorporate the two: like add the `YamlPropertySourceLoader` class to the loader types if the `yaml.to.java` property is present.